src/crystal/system/win32/event_loop_iocp.cr

```diff
@@ -1,14 +1,22 @@
 require "crystal/system/print_error"
 
 module Crystal::EventLoop
-  @@queue = Deque(Fiber).new
+  @@queue = Deque(Event).new
 
   # Runs the event loop.
   def self.run_once : Nil
-    next_fiber = @@queue.pop?
+    next_event = @@queue.min_by { |e| e.wake_at }
 
-    if next_fiber
-      Crystal::Scheduler.enqueue next_fiber
+    if next_event
+      sleep_time = next_event.wake_at - Time.monotonic
+
+      if sleep_time > Time::Span.zero
+        LibC.Sleep(sleep_time.total_milliseconds)
```

Comment:
Hmm. This can only work when all events are sleeps, right? 

Otherwise a completion could happen but not be received until the sleep is over. I think the fetch completion command has a timeout that can be used for the case when there is both completions and sleeps happening. But perhaps actual sleep would still be necessary in the case where there are no completions, that depends on how the fetching command behave if empty.

Reply:
Yes, this will later be changed to `GetQueuedCompletionStatus`. See comment in the OP. For now, `Sleep` is good enough until we can introduce completion ports properly (this is a preparatory step for that).